mongo/src/main/resources/js/populate-hidden-fields.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * 
+ */
+{
+    function populateHiddenFields(fieldMap) {
+        var cursor = db.collection.find();
+        cursor.forEach(function(doc) {
+            // fieldMap is <index, hiddenField>
+            for (var k in fieldMap) {
+                // check if there's an array in the index
+                var arrayIndex = k.lastIndexOf("*");
+                if (arrayIndex > -1) {
+                    doArrayMap(doc, k, fieldMap[k], arrayIndex);
+                } else {
+                    // if no array, just do a direct update
+                    doc[fieldMap[k]] = doc[k].toUpperCase();
```

Comment:
What do you mean by that?

This should be able to handle updating multi-nested arrays.